### ISSUE TITLE
Number of subscriptions

### ISSUE THREAD

**Issue by jc211** [eclipse-cyclonedds/cyclonedds-python#278]
Is there a way to quickly find out if a reader or a writer has been matched without using a WaitSet or a listener, and if so how many subscribers are on the other end? And possibly who they are?

**Comment by eboasson** [comment]
Yes, it is possible. For a reader you can find out about matching writers:

* `dds_get_subscription_matched_status` gives you number of matched writers
* `dds_get_matched_publications` gives you a list of "instance handles" of matching writers
* `dds_get_matched_publication_data` gives you details on a given writer

A similar trio of functions exists for a writer (the names change in the obvious way).

The `dds_get_..._status` call is really cheap (esp. when compared to the other two), so if you want to poll at a high frequency, use that one to decide whether or not to look more closely at the details.

**Comment by eboasson** [comment]
I wasn't paying enough attention, it seems. You created an issue on the python repository, so presumably you want to do it in Python, not in C 🤦

It seems those functions are not (currently) available in the Python binding. It should be a straightforward addition, so if I am really lucky you're willing to try adding it and doing a PR. Otherwise, I (or someone else) can probably find some time in the near future ...

**Comment by jc211** [comment]
I can give it a shot, but maybe you can point me in the right direction. What is the mechanism used here for creating the python bindings. It doesn't seem to be pybind (which I am most familiar with)?

**Comment by jc211** [comment]
For example: to add ```dds_get_subscription_matched_status```, would I add it after     [@c_call("dds_reader_wait_for_historical_data")](https://github.com/eclipse-cyclonedds/cyclonedds-python/blob/2b5231f620b9643b6651dc9647eb5dfbbd4fec0b/cyclonedds/sub.py#L386).

It would be

```python
    @c_call("dds_get_subscription_matched_status")
    def _get_subscription_matched_status(self, reader: dds_c_t.entity, status: dds_c_t.status) -> dds_c_t.returnv:
        pass
```

I don't think dds_c_t.status exists though. Maybe its just an int?

**Comment by jc211** [comment]
It actually looks like someone has done this already: https://github.com/eclipse-cyclonedds/cyclonedds-python/pull/43/commits/da14e5b17004e1924e22b727ca916439a595dff6

**Comment by eboasson** [comment]
> It actually looks like someone has done this already: [da14e5b](https://github.com/eclipse-cyclonedds/cyclonedds-python/commit/da14e5b17004e1924e22b727ca916439a595dff6)

Oh dear. Now I really feel terrible.

Let me see if I can revive that effort, the basis of the binding hasn't changed so presumably it is still pretty close to what's needed.
